### PR TITLE
Support `extern_crate_self` syntax

### LINKED ISSUE #3287: New syntax (rust 1.34) - extern crate self as _
See https://github.com/rust-lang/rust/issues/56409
And https://github.com/rust-lang/rust/pull/57407

---

```rust
extern crate self as hello;

// following two lines are identical
hello::ident
crate::ident
```

### PR DESCRIPTION
**Pull request by Undin**
* Parse `extern_crate_self` syntax
* Implement name resolution
* Highlight errors related to new `extern_crate_self` syntax

Closes #3287

### CODE REVIEW

File: src/test/kotlin/org/rust/lang/core/resolve/RsResolveTest.kt [hunk 1]
```diff
@@ -1193,4 +1193,24 @@ class RsResolveTest : RsResolveTestBase() {
...
+
+        struct Foo;
+              //X
+
+        use foo::Foo;
+                //^
+    """)
+
+    fun `test extern crate self without alias`() = checkByCode("""
+        extern crate self;
```
Comment (vlad20012) [inline src/test/kotlin/org/rust/lang/core/resolve/RsResolveTest.kt]: Looks like this test don't test the feature b/c it should pass without `extern crate self;`

Comment (vlad20012) [reply src/test/kotlin/org/rust/lang/core/resolve/RsResolveTest.kt]: Oh, or it tests that behavior of `self` isn't changed by adding (wrong) `extern crate self;`

Comment (Undin) [reply src/test/kotlin/org/rust/lang/core/resolve/RsResolveTest.kt]: > Oh, or it tests that behavior of self isn't changed by adding (wrong) extern crate self;

Yep

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**